nonship Jakarta NoSQL/Data features

njr-11 opened the pull request:
Create a nonship Jakarta NoSQL feature by pulling in the latest Jakarta NoSQL artifacts from Maven and using the transformer to switch them to jakarta package names.  This will help us in exploring how the design of Jakarta Data can interact with and reuse from the Jakarta NoSQL API.

njr-11 committed:
Issue #22443 nonship Jakarta NoSQL

njr-11 committed:
Issue #22443 start of NoSQL test bucket for Jakarta Data

njr-11 commented:
#build

njr-11 commented:
#build
#libby

njr-11 commented:
#build
#libby

njr-11 committed:
Issue #22443 fix or work around feature issues

njr-11 commented:
#build

helyarp commented:
I added the **ID review not needed** label because the .properties files have no messages to review. The strings in the .properties files look fine.

posmith commented:
L2 message review not required

njr-11 commented:
#build
#libby

njr-11 committed:
Issue #22443 add translatable description to prevent build failure

njr-11 commented:
#build
#libby

njr-11 commented:
#build
#libby

nmittles commented on dev/io.openliberty.jakarta.data.1.0/src/jakarta/data/Delete.java:
```diff
@@ -0,0 +1,28 @@
+import java.lang.annotation.Target;
+
+/**
+ * Annotates a repository method with to designate it as a delete operation.
```
Check grammar.

nmittles commented on dev/io.openliberty.jakarta.data.1.0/src/jakarta/data/Delete.java:
```diff
@@ -0,0 +1,28 @@
+ * Annotates a repository method with to designate it as a delete operation.
+ * The <code>@Where</code> annotation can be added to provide conditions.
+ * Do not combine on a single method with {@link Update &#64;Update} or with
+ * {@link Query &#64;Query}, which is can be more advanced way of providing
```
Check grammar.

njr-11 committed:
Issue #22443 nonship Jakarta Data feature

njr-11 commented:
Fixed review comment and a bundle name typo.
#build
#libby

njr-11 committed:
Issue #22443 more corrections to unused feature that is only needed to allow for transform

njr-11 commented:
Rebased on to latest and fixed issue with a nonship dummy feature that we only need to allow transform so that the unused original bundle won't generate complaints
#build
#libby

njr-11 commented:
#build
#libby

njr-11 committed:
adjustments for EE compatibility tests

njr-11 commented:
#build
#libby

njr-11 commented:
The build needed to have 1 child FAT resubmitted due to what looks like an infrastructure issue, and then the resubmit got stuck but in a different place, but otherwise looks good.
